File: frontend/src/types/category.ts
export interface Category {
  id: string;
  name: string;
  slug: string;
  description: string | null;
  parent_id: string | null;
  attribute_schema: Record<string, unknown>;
  sort_order: number;
  created_at: string;
  updated_at: string;
}

export interface CategoryTree extends Category {
  children: CategoryTree[];
}

export interface CategoryCreate {
  name: string;
  slug: string;
  description?: string;
  parent_id?: string;
  attribute_schema?: Record<string, unknown>;
  sort_order?: number;
}

export interface CategoryUpdate {
  name?: string;
  slug?: string;
  description?: string | null;
  parent_id?: string | null;
  sort_order?: number;
}

File: backend/app/schemas/category.py
from datetime import datetime

from pydantic import BaseModel


class CategoryCreate(BaseModel):
    name: str
    slug: str
    description: str | None = None
    parent_id: str | None = None
    attribute_schema: dict = {}
    sort_order: int = 0


class CategoryUpdate(BaseModel):
    name: str | None = None
    slug: str | None = None
    description: str | None = None
    parent_id: str | None = None
    attribute_schema: dict | None = None
    sort_order: int | None = None


class CategoryRead(BaseModel):
    id: str
    name: str
    slug: str
    description: str | None
    parent_id: str | None
    attribute_schema: dict
    sort_order: int
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class CategoryTree(CategoryRead):
    children: list["CategoryTree"] = []
